extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcBasePersistenceImpl.java

```diff
@@ -200,6 +203,31 @@ public void writeToGrantRecords(
     }
   }
 
+  @Override
+  public void writeEvents(@Nonnull List<PolarisEvent> events) {
+    int batchSize = 10;
+    List<List<Converter<PolarisEvent>>> batchedModelEvents = new ArrayList<>();
```

Comment:
It seems like you could simplify this a lot by not pre-partitioning. Idiomatically this is something like:


```
ArrayList<PolarisEvent> batch = new ArrayList(batchSize);
for (PolarisEvent event : events) {
  batch.add(event)
  if (batch.size() == batchSize) {
    statement.executeUpdate(generateMultipleInsertQuery(batch, realmId));
    batch.clear()
  }
}
if (batch.size() > 0) {
  statement.executeUpdate(generateMultipleInsertQuery(batch, realmId));
}
```

If you're okay with Guava, you can just do something like:
```
List<List<PolarisEvent>> partitions = Lists.partition(events, batchSize);
for (List<PolarisEvent> batch : partitions) {
    statement.executeUpdate(generateMultipleInsertQuery(batch, realmId));
}
```